### PR TITLE
set /go/src/github.com/cockroachdb/cockroach volume properly

### LINKED ISSUE #3575: A volume is not set properly for cockroachdb/builder docker image
`builder.sh` and `build-docker-deploy.sh` do not set `/go/src/github.com/cockroachdb/cockroach` volume properly for `cockroachdb/builder` docker image if they are not launched from `cockroach` directory. The following text shows the effect if they are launched from `cockroach/build` directory.

In `cockroach` directory, run the following commands to launch `build-docker-deploy.sh` :

```
> cd build
> ./build-docker-deploy.sh 
make: *** No rule to make target 'release'.  Stop.
```

If `builder.sh` is launched from `cockroach/build` folder, `/go/src/github.com/cockroachdb/cockroach` will be mapped to `cockroach/build` directory instead of `cockroach` directory.

### PR DESCRIPTION
**Pull request by yaojingguo**
Fixed #3575

[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/3576)

### CODE REVIEW

File: build/build-docker-deploy.sh [hunk 1]
```diff
@@ -24,7 +30,7 @@ if [ "${1-}" = "docker" ]; then
 fi
 
 # Build the cockroach and test binaries.
-$(dirname $0)/builder.sh $0 docker
+$(dirname $0)/builder.sh $(path_for_docker) docker
```
Comment (tamird) [inline build/build-docker-deploy.sh]: Why are these changes needed?

File: build/builder.sh [hunk 2]
```diff
@@ -8,6 +8,11 @@ function init() {
   docker build --tag="${image}" "$(dirname $0)"
 }
 
+function cockroach_dir() {
+  local dir=$(cd $(dirname $0); pwd)
```
Comment (tamird) [inline build/builder.sh]: Why the `cd` gymnastics? `$(dirname $(dirname $0))` should suffice

Comment (petermattis) [inline build/builder.sh]: If you invoke the script as `build/builder.sh`, `$(dirname build/builder.sh)` will return `build` and `$(dirname build)` will return `.`.

File: build/build-docker-deploy.sh [hunk 3]
```diff
@@ -7,6 +7,12 @@ set -euo pipefail
 
 source $(dirname $0)/build-common.sh
 
+function path_for_docker() {
```
Comment (petermattis) [inline build/build-docker-deploy.sh]: I'm not that fond of this function name. Rather than a function, you could just declare a global variable:

```
# Relative path from the top-level cockroach directory to the current script.
relative_path_to_self="$(basename $(cd $(dirname $0); pwd))/$(basename $0)"
```

This also removes any concern about whether `$0` refers to script args or function args. (I would have thought script args, so I'm not actually sure how this is working).

File: build/build-docker-deploy.sh (hunk 1 above)
Comment (petermattis) [inline build/build-docker-deploy.sh]: They allow the script to be independent of the current directory. If not these changes, the script should check that the current directory is the top-level cockroach directory which is the only place it currently works.

File: build/builder.sh (hunk 2 above)
Comment (petermattis) [inline build/builder.sh]: Per my other comment, I think this would be better as a global variable:

```
# Absolute path to the toplevel cockroach directory.
cockroach_toplevel="$(dirname $(cd $(dirname $0); pwd))"
```

**Comment by petermattis** [comment]
Thanks for making this script a bit more robust.

File: build/build-docker-deploy.sh (hunk 1 above)
Comment (yaojingguo) [inline build/build-docker-deploy.sh]: If `build-docker-deploy.sh` is launched with `./build-docker-deploy.sh` from  `cockroach/build` directory. $0 will be `./build-docker-deploy.sh` which does not works since workdir for the docker image is mapped to `cockroach` directory. `build-docker-deploy.sh` exists in `cockroach/build` directory not in `cockroach` directory.

**Comment by yaojingguo** [comment]
@petermattis, a new commit after your suggested changes.

**Comment by petermattis** [comment]
LGTM, though squash before committing.
